### PR TITLE
(challenge 1):  Implement Todo's endpoints

### PR DESCRIPTION
**Pull request by adex001**
#### What does this PR do?
- Creates all todos endpoints and their corresponding tests
#### Description of Task to be completed?
- write test to `GET /api/v1/to-do` endpoint and implement its functionality to retrieve all todos
- write test to `GET /api/v1/to-do/<to-do>` endpoint and implement its functionality to retrieve a specific todo
- write test to `PATCH /api/v1/to-do/<to-do>` endpoint and implement its functionality to update a specific todo
- write test to `DELETE /api/v1/to-do/<to-do>` endpoint and implement its functionality to delete a specific todo
- write test to `POST /api/v1/to-do` endpoint and implement its functionality to create a new todo

#### How should this be manually tested?
- Open `Postman` and run the above endpoints
#### Any background context you want to provide?
- N/A
#### What are the relevant pivotal tracker stories?
- N/A
#### Screenshots (if appropriate)
- N/A
#### Questions:
- N/A

### CODE REVIEW

File: model/todo.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        return single_todo
+    
+    def create_todo(self, todo_object):
+        todos.insert(0, todo_object)
+        return todo_object
+
+
+    def delete_todo(self, todo_id):
+        todo_id = int(todo_id, base=10)
+        i = 0
```
Comment (Hesbon5600) [inline model/todo.py]: Awesome job @adex001 You don't have to use `i` to loop through. `for` will loop through the object

**Comment by Hesbon5600** [comment]
Update your PR description to explain how to test the PR

**Comment by Hesbon5600** [comment]
Also, your commit message does not follow the commit message convention

File: views/todo.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+    def patch(self, todo_id):
+        todo_object = request.get_json()
+        if (todo_object is None) or ('title' not in todo_object) and ('body' not in todo_object) and ('completed' not in todo_object):
+            return jsonify({'error': 'Enter either a title, body or completed'})
+        if (not TodoModel().search_todo(todo_id)):
+            return jsonify({'error': 'Todo not found'})
+        todo = TodoModel().update_todo(todo_id, todo_object)
+        return jsonify({'todo': todo})
+
```
Comment (Nyakaru) [inline views/todo.py:60]: Please add a new line here

**Review by Nyakaru — COMMENTED** [review]
Please also add some simple test i.e `to even assert status codes`

File: model/todo.py (hunk 1 above)
Comment (adex001) [reply model/todo.py]: alright. Thanks for your feedback. But is there any other way to get the index so I can use the pop method?

Comment (Hesbon5600) [reply model/todo.py]: Try `todos .remove(todo)`

Comment (adex001) [reply model/todo.py]: ok! I will.. Thanks!

**Pull request merged**